sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -603,6 +603,8 @@ private void validateUserInputForKeyAndValue() throws IOException, InterruptedEx
      * Returns parameterized type of the InputFormat class.
      */
     private ParameterizedType determineGenericType() {
+      // It is considered that user provided inputFormatClass will always inherit from
+      // ParameterizedType.
```

Comment:
this doesn't really help me understand *why* we assume that's the case.  What is the reason you know that will be true? what property of InputFormat?